src/mlpack/methods/lsh/lsh_search_impl.hpp

```diff
@@ -822,13 +861,13 @@ void LSHSearch<SortPolicy>::Search(const arma::mat& querySet,
 
     // An informative book-keeping for the number of neighbor candidates
     // returned on average.
+    // Make atomic to avoid race conditions when multiple threads are running
+    #pragma omp atomic
     avgIndicesReturned += refIndices.n_elem;
```

Comment:
Could we use a reduction here in the parallel for definition?  i.e. `reduction(+,avgIndicesReturned)`.  Then you wouldn't need an atomic here.  It might speed things up (not sure how much).